File: bench/Reference/file_parallel_read.js
const fs = require('fs').promises;
const path = 'bench/data/bench_text.txt';
Promise.all([
    fs.readFile(path, 'utf8'),
    fs.readFile(path, 'utf8'),
    fs.readFile(path, 'utf8'),
]).then(results => {
    console.log(results.reduce((a, s) => a + s.length, 0));
});

File: bench/Reference/file_parallel_read.erl
#!/usr/bin/env escript
%% Read the same file three times in parallel; sum byte lengths.
%% Uses processes + a receive loop for the join.

main(_) ->
    Parent = self(),
    _ = [spawn_link(fun() ->
             {ok, B} = file:read_file("bench/data/bench_text.txt"),
             Parent ! {self(), byte_size(B)}
         end) || _ <- [1, 2, 3]],
    Total = collect(3, 0),
    io:format("~p~n", [Total]).

collect(0, Acc) -> Acc;
collect(N, Acc) ->
    receive
        {_, Sz} -> collect(N - 1, Acc + Sz)
    end.
